app/src/main/java/org/openobservatory/ooniprobe/activity/MainActivity.java

```diff
@@ -181,55 +187,122 @@ private void requestNotificationPermission() {
     @Override
     protected void onNewIntent(Intent intent) {
         super.onNewIntent(intent);
-        if (intent.getExtras() != null){
-            if (intent.getExtras().containsKey(RES_ITEM))
-                binding.bottomNavigation.setSelectedItemId(intent.getIntExtra(RES_ITEM, R.id.dashboard));
-            else if (intent.getExtras().containsKey(NOTIFICATION_DIALOG)){
-                new ConfirmDialogFragment.Builder()
-                        .withTitle(intent.getExtras().getString("title"))
-                        .withMessage(intent.getExtras().getString("message"))
-                        .withNegativeButton("")
-                        .withPositiveButton(getString(R.string.Modal_OK))
-                        .build().show(getSupportFragmentManager(), null);
+        if (Intent.ACTION_VIEW.equals(intent.getAction())) {
+            Uri uri = intent.getData();
+            if (uri == null) return;
+
+            String host = uri.getHost();
+            long runId = 0;
+
+            try {
+                if ("runv2".equals(host)) {
+                    runId = Long.parseLong(uri.getPathSegments().get(0));
+                } else if ("run.test.ooni.org".equals(host)) {
+                    runId = Long.parseLong(uri.getPathSegments().get(1));
+                }
+            } catch (Exception e) {
+                e.printStackTrace();
```

Comment:
Please, add a comment mentioning we handle all errors below.

Also, do we want to document that this exception occurred using third-party services?